### PR TITLE
[Catalog] Render configurable swatches based on the widget's configuration

### PR DESCRIPTION
**Pull request by eduard13**
### Description (*)
This PR fixes the configurable swatches rendering based on the widget's configuration. 

### Fixed Issues (if relevant)

1. magento/magento2#14134: Swatches not showing on homepage New products widget
2. ...

### Manual testing scenarios (*)
1. Create a configurable product with color swatches
2. Use `New Products` widget for showing it on homepage or any other place

**Grid Mode**
xnip2019-02-27_18-53-53

**List Mode**
xnip2019-02-27_18-55-38

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @eduard13. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Catalog/Block/Product/Widget/NewWidget.php [hunk 1]
```diff
@@ -5,6 +5,10 @@
  */
 namespace Magento\Catalog\Block\Product\Widget;
 
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Serialize\Serializer\Json;
+use Magento\Swatches\Block\Product\Renderer\Listing\Configurable;
```
Comment (rogyar) [inline app/code/Magento/Catalog/Block/Product/Widget/NewWidget.php:10]: We should avoid bringing an additional hard dependency between Catalog and Swatches module. The main idea is keeping modules decoupled with as fewer dependencies as possible. 
Thank you

**Review by rogyar — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/Block/Product/Widget/NewWidget.php (hunk 1 above)
Comment (eduard13) [reply app/code/Magento/Catalog/Block/Product/Widget/NewWidget.php:10]: The hard dependency was replaced by injecting the needed block via layout, could you please have a look?
Please note that I used the `root` container for adding the needed block, in order to have it working even if you use the widget in sidebar, content or any other custom container.
Thank you.

Comment (rogyar) [reply app/code/Magento/Catalog/Block/Product/Widget/NewWidget.php:10]: Hi @eduard13. We still have a hard dependency in place (Catalog -> Swatches). We bring it by using `use Magento\Swatches\Block\Product\Renderer\Listing\Configurable;`. That means, if we disable Swatches module our Catalog will not work. Situations like this should be avoided. If there's an issue with swatches we need to figure out how to fix it in swatches without bringing a new dependency to the Catalog module, if possible

**Comment by eduard13** [comment]
Closing this one. Will get back to it with a better implementation.

**Pull request closed without merging**